standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -4664,6 +4664,58 @@ public void dropStoredProcedure(StoredProcedureRequest request) throws MetaExcep
     return client.get_all_stored_procedures(request);
   }
 
+  /**
+   * Builder for the GetProjectionsSpec. This is a projection specification for partitions returned from the HMS.
+   */
+  public static class GetPartitionProjectionsSpecBuilder {
```

Comment:
Can we create a common interface implementing build and have both of the classes implement it?
Is it possible to move the builder implementations to separate file? HiveMetaStoreClient is already too big.

Reply:
From a user perspective, I think it would be more user friendly to have builder class for the whole request objects (eg. GetPartitionsRequestBuilder which gives ability to create a projection and filter spec)

Reply:
@vineetgarg02 

Thanks a lot for taking time to review the changes and comment on them.

The builder classes follow the builder pattern who's definition generally calls for the presence of the build() method. I think creating a separate interface with just the build() method that these classes implement carries a slight risk of making them superfluous. I humbly ask you to reconsider this.

I agree that it makes sense to move the builder implementations to a separate file and ease the burden on the large HiveMetaStoreClient file. But I think the change should not be done as part of this JIRA. Do you think I can create a separate JIRA for this and do it there ?

@vihangk1 

Thank you very much for taking the time to review the changes and comment on them.

I agree that we should look at creating builder methods for GetTableRequest and GetPartitionRequest. But I am inclined to push the request builders out of this JIRA. Do you think it would be OK to retain the projection specification builders in this JIRA, create a separate JIRA for the Request builder and handle that once I am done with the parent issue ?